packages/slice-machine/src/features/sync/actions/trackPushChangesSuccess.ts

```diff
@@ -94,4 +96,82 @@ export function trackPushChangesSuccess(args: TrackPushChangesSuccessArgs) {
     hasDeletedDocuments,
     _includeEnvironmentKind: true,
   });
+
+  trackPushedGroups({ changedCustomTypes, changedSlices });
+}
+
+type TrackPushedGroupsArgs = {
+  changedSlices: ReadonlyArray<ChangedSlice>;
+  changedCustomTypes: ReadonlyArray<ChangedCustomType>;
+};
+
+type FieldStats = {
+  isInStaticZone: boolean;
+  isInSlice: boolean;
+} & {
+  [key in FieldType]?: number;
+};
+
+function trackPushedGroups(args: TrackPushedGroupsArgs) {
+  const { changedCustomTypes, changedSlices } = args;
+
+  const groupsInStaticZone = changedCustomTypes.reduce<FieldStats[]>(
+    (acc, customType) => {
+      if (customType.status !== ModelStatus.New) return acc;
+
+      customType.customType.tabs.forEach((tab) => {
+        tab.value.forEach((field) => {
+          if (field.value.type === "Group" && field.value.config?.fields) {
+            const fieldStats: FieldStats = {
+              isInStaticZone: true,
+              isInSlice: false,
+            };
+            field.value.config.fields.forEach((field) => {
+              if (typeof fieldStats[field.value.type] !== "number") {
+                fieldStats[field.value.type] = 1;
+              } else {
+                fieldStats[field.value.type]! += 1;
+              }
+            });
+            acc.push(fieldStats);
+          }
```

Comment:
💡 This logic seems to be the same as the one below, so it could be extracted to a function.

Reply:
💡 It could also be simplified to something like this, to avoid non-null assertions

```suggestion
          if (field.value.type === "Group" && field.value.config?.fields) {
            const groupCount: { [key in FieldType]?: number } = {};
            field.value.config.fields.forEach(({ value: fieldValue }) => {
              const value = groupCount[fieldValue.type] ?? 0;
              groupCount[fieldValue.type] = value + 1;
            });
            acc.push({ isInStaticZone: false, isInSlice: true, ...groupCount });
          }
```

Reply:
Nice, I'll try that, was already trying to extract that duplicated part but it got super confusing with types very quickly, but will try another attempt.

Reply:
I added your suggestion, but didn't extract repeated logic to another function as correct typing for that function would be very convoluted.

Reply:
Would be something like this:

```ts
customType.customType.tabs.forEach((tab) => {
  const groupFields = countGroupFields(acc, tab.value);
  acc.push(...groupFields);
});

// ...

slice.slice.model.variations.forEach((variation) => {
  const groupFields = countGroupFields(acc, variation.primary);
  acc.push(...groupFields);
});


// ...

interface Field {
  key: string;
  value: GroupSM | UID | NestableWidget;
}

function countGroupFields(
  fieldStats: FieldStats[],
  fields: Field[] | undefined,
) {
  if (!fields) return fieldStats;

  const newFieldState = fieldStats.slice();
  fields.forEach((field) => {
    if (field.value.type === "Group" && field.value.config?.fields) {
      const fieldsCount: FieldsCount = {};
      field.value.config.fields.forEach(({ value: fieldValue }) => {
        const value = fieldsCount[fieldValue.type] ?? 0;
        fieldsCount[fieldValue.type] = value + 1;
      });
      newFieldState.push({
        isInStaticZone: false,
        isInSlice: true,
        ...fieldsCount,
      });
    }
  });

  return newFieldState;
}
```

Reply:
nice! applied your suggestion!